master/internal/core.go

```diff
@@ -218,45 +218,62 @@ func (m *Master) getRawResourceAllocation(c echo.Context) error {
 
 func (m *Master) getAggregatedResourceAllocation(c echo.Context) error {
 	args := struct {
-		Start string `query:"start_date"`
-		End   string `query:"end_date"`
+		Start  string `query:"start_date"`
+		End    string `query:"end_date"`
+		Period string `query:"period"`
 	}{}
 	if err := api.BindArgs(&args, c); err != nil {
 		return err
 	}
 
-	start, err := time.Parse("2006-01-02", args.Start)
-	if err != nil {
-		return errors.Wrap(err, "invalid start date")
-	}
-	end, err := time.Parse("2006-01-02", args.End)
-	if err != nil {
-		return errors.Wrap(err, "invalid end date")
-	}
-	if start.After(end) {
-		return errors.New("start date cannot be after end date")
-	}
-
 	resp := &apiv1.ResourceAllocationAggregatedResponse{}
-	if err := m.db.QueryProto(
-		"get_aggregated_allocation", &resp.ResourceEntries, start.UTC(), end.UTC(),
-	); err != nil {
-		return errors.Wrap(err, "error fetching allocation data")
-	}
 
-	c.Response().Header().Set("Content-Type", "text/csv")
+	switch args.Period {
+	case "RESOURCE_ALLOCATION_AGGREGATION_PERIOD_DAILY":
```

Comment:
Could you convert args.Period to a `masterv1.ResourceAllocationAggregationPeriod` and then de-dup this code with the code in `api_master.go`?

Reply:
Great idea, done!